doc/source/getting_started.rst

```diff
@@ -0,0 +1,113 @@
+attributes (e.g., description, unit, parent region).
+
+A **RegionProcessor** holds a list of RegionAggregationMappings and a
+DataStructureDefinition. This class is used to facilitate region processing for model
+comparison studies.
+
+A **RegionAggregationMapping** is a mapping that defines on a per-model basis how model
+native regions should be renamed and aggregated to comparison regions.
+
+
+Minimum working example
+-----------------------
+
+This section aims to provide a minimum working example of how to use the nomenclature
+package. It is assumed that the variable templates and model mappings already exist.
+Details on how those are structured and how to start using nomenclature "from scratch"
+can be found here :ref:`usage`. 
+
+The following outlines how to use the nomenclature package:
+
+.. code-block:: python
+
+   # Import the necessary libraries
+   import pyam
+   from nomenclature import DataStructureDefinition, RegionProcessor
+   
+   # Initialize DataStructureDefinition and RegionProcessor giving them the
+   # directories where the codelists and mappings are defined as input.
+   dsd = DataStructureDefinition("definitions/")
+   rp = RegionProcessor.from_directory("mappings/", dsd)
+
+   # Read in the data using pyam
+   iam_results_file = "some file"
+   df = pyam.IamDataFrame(iam_results_file)
+
+   # Validate that the data frame only contains allowed regions and variables
+   dsd.validate(df)
+   # Apply region processing to the data
+   df = rp.apply(df)
+
+**Notes**
+
+* The pyam library is required as *DataStructureDefinition.validate()* and
+  *RegionProcessor.apply()* take a *pyam.IamDataFrame* as input.
+
+* *DataStructureDefinition* and *RegionProcessor* are initialized from directories
+  containing yaml files. See :ref:`dir-structure` for details. 
+
+* *DataStructureDefinition.apply()* returns none if the data frame only contains   
+  allowed values and raises an error otherwise.
+
+
+.. _dir-structure:
+
+Directory structure for definitions and mappings
+------------------------------------------------
+
+This is the directory structure that needs to be in place in order for the validation and region processing to work:
+
+.. code-block:: bash
+
+   .
+   ├── definitions
+   │   ├── region
+   │   │   ├── ...
+   │   │   └── regions.yaml
+   │   └── variable
+   │       ├── ...
+   │       └── variable.yaml
+   └── mappings
+       ├── model_a.yaml
+       └── ...
+
+**Notes**
+
+* The names of the *definitions* and *mappings* directories are purely conventional and
+  don't carry any special meaning. As long as *DataStructureDefinition* and
+  *RegionProcessor.from_directory()* are pointed to the correct directories everything
+  will work.
+
+* Inside the *definitions* directory each "dimension", in our case *variable* and
+  *region*, must live in its own sub-directory.
+
+* As opposed *definitions* and *mappings*, *variable* and *region* are special names
```

Comment:
I think the tests will fail if you don't have a "definitions" folder.

Reply:
Do you mean the unit tests or the cli? I just took a look and you're right the cli `assert_valid_structure` would fail as it assumes that 'definitions' and 'mappings' are in place.
This is an inconsistency, though as for the actual classes `DataStructureDefinition` and `RegionProcessor` the constructors take a path argument that can be a directory of any name, not just the fixed 'definitions' and 'mappings'. 
I would vote to fix that inconsistency by either:

1. Changing the constructors to only look for these fixed directory names.
2. Add two optional arguments to the CLI to specify the names of the definitions and mappings directories is they are non-standard. The default values for the parameters would be 'definitions' and 'mappings' of course so it's transparent to the user.

I personally would prefer option 2.

Reply:
Same. @danielhuppmann what do you think?

Reply:
No reason against extending the testing-module - but I still push users into adopting the conventions, see other suggestions.

Reply:
Sounds good, I've updated the docs.